pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v1/PersistentTopics.java

```diff
@@ -462,7 +462,47 @@ public void expireTopicMessages(@Suspended final AsyncResponse asyncResponse,
             @QueryParam("authoritative") @DefaultValue("false") boolean authoritative) {
         try {
             validateTopicName(property, cluster, namespace, encodedTopic);
-            internalExpireMessages(asyncResponse, decode(encodedSubName), expireTimeInSeconds, authoritative);
+            internalExpireMessagesByTimestamp(asyncResponse, decode(encodedSubName),
+                    expireTimeInSeconds, authoritative);
+        } catch (WebApplicationException wae) {
+            asyncResponse.resume(wae);
+        } catch (Exception e) {
+            asyncResponse.resume(new RestException(e));
+        }
+    }
+
+    @POST
+    @Path("/{tenant}/{namespace}/{topic}/subscription/{subName}/expireMessages")
```

Comment:
I remember in V1 APIs, the topic name is constructed with tenant, cluster, namespace, and topic name. It's a little different with V2. You can check the other APIs, they are carrying the cluster name in the request path.

Do we still need to update the v1 Rest API? @sijie @codelipenghui

Reply:
If we are going to update the v1 reset API, we should follow the v1 pattern.

Reply:
oh sure, will update.